### PR TITLE
Issue 359 filter metadata date marked as complete

### LINKED ISSUE #359: Can't filter by metadata field date_marked_as_complete
#### Software and hardware versions 
Aggregate v1.?.? (since 2011)

#### Problem description
The server fails to filter submissions by the metadata field "date_marked_as_complete".

#### Steps to reproduce the problem
Show metadata fields in the submissions list and try to add a filter using this field.

#### Expected behavior
It should work

#### Other information 
There's a problem with a `switch` block in `QueryBase`. The `case` for this field is missing the `break`;

### PR DESCRIPTION
**Pull request by ggalmazor**
Closes #359, and #360

#### What has been done to verify that this works as intended?
Try to filter submissions by submission date or "date marked as complete" metadata fields and verify it works

#### Why is this the best possible solution? Were any other approaches considered?
This is the narrowest solution I could thin of.

#### Are there any risks to merging this code? If so, what are they?
Nope.

#### Do we need any specific form for testing your changes? If so, please attach one
No.

#### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.
No.

### CODE REVIEW

File: src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java [hunk 1]
```diff
@@ -135,50 +135,72 @@ public QueryByUIFilterGroup(IForm form, FilterGroup filterGroup, CompletionFlag
...
-            continue;
+            break;
           case JRTIME:
-            compareValue = WebUtils.parseDate(value);
             super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
-            continue;
+            break;
           case INTEGER:
-            compareValue = Long.valueOf(value);
+            super.addFilter(fem, op, compareValue);
```
Comment (dcbriccetti) [inline src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java]: How about grouping together these four cases so this only appears once:
`super.addFilter(fem, op, compareValue);`

File: src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java [hunk 2]
```diff
@@ -135,50 +135,72 @@ public QueryByUIFilterGroup(IForm form, FilterGroup filterGroup, CompletionFlag
...
       }
     }
 
   }
 
+  public Object getCompareValue(FormElementModel fem, String value) {
+    switch (fem.getElementType()) {
+      case BOOLEAN:
+        return WebUtils.parseBoolean(value);
+      case JRDATETIME:
```
Comment (dcbriccetti) [inline src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java]: Same suggestion here about consolidating like cases

File: src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java [hunk 3]
```diff
@@ -139,6 +139,10 @@ public QueryByUIFilterGroup(IForm form, FilterGroup filterGroup, CompletionFlag
 
         switch (fem.getElementType()) {
           case BOOLEAN:
+          case INTEGER:
+          case DECIMAL:
+          case SELECT1:
+          case STRING:
```
Comment (dcbriccetti) [inline src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java]: Much cleaner!

Comment (ggalmazor) [reply src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java]: Yup, you have an eye for this :) Thanks!

**Pull request merged**